enterprise/frontend/src/embedding/data-picker/DataSelector/tests/DataSourceSelector.unit.spec.tsx

```diff
@@ -0,0 +1,185 @@
+import userEvent from "@testing-library/user-event";
+import fetchMock from "fetch-mock";
+
+import { setupDatabasesEndpoints } from "__support__/server-mocks";
+import { renderWithProviders, screen } from "__support__/ui";
+import { getNextId } from "__support__/utils";
+import type { Database, Table } from "metabase-types/api";
+import { createMockDatabase, createMockTable } from "metabase-types/api/mocks";
+import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+import {
+  createMockSettingsState,
+  createMockState,
+} from "metabase-types/store/mocks";
+
+import { DataSourceSelector } from "../DataSelector";
+
+const DATABASES = [createSampleDatabase()];
+
+const storeInitialState = createMockState({
+  settings: createMockSettingsState({
+    "enable-nested-queries": true,
+  }),
+});
+
+const AVAILABLE_MODELS: Record<AvailableModels, ("dataset" | "table")[]> = {
+  "tables-only": ["table"],
```

Comment:
there's no `models-only` situation because models are created from `table` so if `dataset` is available, it means `table` is also available.